api/src/main/java/org/openmrs/module/metrics/api/db/hibernate/HibernateMetricsDAO.java

```diff
@@ -35,34 +37,44 @@ public MetricEvent saveMetricEvent(MetricEvent metricEvent) {
 	@Override
 	public Integer getEncounterObjectsByGivenDateRangeAndType(LocalDateTime startRange, LocalDateTime endRange,
 	        String encounterType) {
-		String sql = "SELECT count(*)"
-		        + "FROM event_records e inner join encounter b on e.object_uuid = b.uuid inner join encounter_type"
-		        + " c on b.encounter_type = c.encounter_type_id" + "WHERE e.title='Encounter' and (e.date_created BETWEEN "
-		        + startRange + " and " + endRange + ") and e.tags = 'CREATED' " + "and c.name = " + encounterType + "";
+		String sql = "select count(*) "
+		        + "from event_records e "
+		        + "inner join encounter b on e.object_uuid = b.uuid "
+		        + "inner join encounter_type c on b.encounter_type = c.encounter_type_id "
+		        + "where e.title = 'Encounter' and (e.date_created between :startRange and :endRange) and e.tags = 'CREATED' "
+		        + "and c.name = :encounterType";
 		
-		Integer count = ((BigInteger) sessionFactory.getCurrentSession().createSQLQuery(sql).uniqueResult()).intValue();
-		return count;
+		return (Integer) sessionFactory.getCurrentSession().createSQLQuery(sql)
+		        .setDate("startRange", Date.from(startRange.atZone(ZoneId.systemDefault()).toInstant()))
+		        .setDate("endRange", Date.from(endRange.atZone(ZoneId.systemDefault()).toInstant()))
+		        .setString("encounterType", encounterType).uniqueResult();
 	}
 	
 	@Override
 	public Integer getNewPatientsObjectsByGivenDateRange(LocalDateTime startRange, LocalDateTime endRange) {
-		String sql = "SELECT count(*)" + "FROM event_records e" + "WHERE e.title='Patient' and (e.date_created BETWEEN "
-		        + startRange + " and " + endRange + ") and e.tags = 'CREATED'";
+		String sql = "select count(*)" + "from event_records e"
```

Comment:
You're actually going to need spaces at the end of these strings. I.e., this will compile down to:

```sql
select count(*)from event_records ewhere e.title='Patient' and (e.date_created between :startRange and :endRange) and e.tags = 'CREATED'"
```